src/kernel/arch/aarch64/boot.zig

```diff
@@ -8,32 +9,41 @@ extern var KERNEL_STACK_END: *usize;
 
 export fn _start() linksection(".text.boot") callconv(.Naked) noreturn {
     // Halt all cores other than the primary core, until the kernel has multicore support
-    const core_id = asm ("mrs %[res], mpidr_el1"
-        : [res] "=r" (-> usize)
-    ) & 3;
-    if (core_id != 0) {
-        while (true) {
-            asm volatile ("wfe");
-        }
-    }
+    asm volatile (
+        \\ mrs x0, mpidr_el1
+        \\ mov x1, #3
+        \\ ands x0, x0, x1
+        \\ beq cpu0
+        \\hang:
+        \\ wfe
+        \\ b hang
+        \\cpu0:
+    );
 
     // Setup the stack
-    asm volatile ("mov sp, %[stack_end]"
+    asm volatile (
+        \\mov sp, %[stack_end]
         :
         : [stack_end] "r" (@ptrCast([*]u8, &KERNEL_STACK_END))
     );
 
     // Setup the exception table
     asm volatile (
-        \\msr vbar_el1, %[table_addr]
-        \\msr vbar_el2, %[table_addr]
-        \\msr vbar_el3, %[table_addr]
+        \\ msr vbar_el1, %[table_addr]
+        \\ msr vbar_el2, %[table_addr]
+        \\ mrs x0, CurrentEL
+        \\ lsr x0, x0, 2
+        \\ and x0, x0, #3
+        \\ cmp x0, #3
+        \\ bne not_el3
```

Comment:
For some reason rpi3b+ actually starts in el2 so el3 is not accessible. Needs investigation.